tools/finish_release.py

```diff
@@ -0,0 +1,199 @@
+        r = requests.get(url)
+        r.raise_for_status()
+        with open(tempdir + '/' + filename + '.zip', 'wb') as f:
+            f.write(r.content)
+        print("Extracting %s" % filename)
+        with ZipFile(tempdir + '/' + filename + '.zip', 'r') as zipObj:
+           zipObj.extractall(tempdir)
+
+    version = build_client.get_build('certbot', build_id).source_branch.split('v')[1]
+    return version
+
+def create_github_release(github_access_token, tempdir, version):
+    """Use build artifacts to create a github release, including uploading additional assets
+
+    :param str github_access_token: string containing github access token
+    :param str path: path to a temporary directory where azure artifacts are located
+    :param str version: Certbot version number, e.g. 1.7.0
+
+    """
+    # Create release
+    g = Github(github_access_token)
+    repo = g.get_user('certbot').get_repo('certbot')
+    release_notes = open(tempdir + '/changelog/release_notes.md', 'r').read()
+    print("Creating git release")
+    release= repo.create_git_release('v{0}'.format(version),
+                                     'Certbot {0}'.format(version),
+                                     release_notes,
+                                     draft=True)
+
+    # Upload windows installer to release
+    print("Uploading windows installer")
+    release.upload_asset(tempdir + '/windows-installer/certbot-beta-installer-win32.exe')
+    release.update_release(release.title, release.body, draft=False)
+
+
+def assert_logged_into_snapcraft():
+    """Confirms that snapcraft is logged in to an account.
+
+    :raises SystemExit: if the command snapcraft is unavailable or it
+        isn't logged into an account
+
+    """
+    cmd = 'snapcraft whoami'.split()
+    try:
+        subprocess.run(cmd, check=True, stdout=subprocess.DEVNULL,
+                       stderr=subprocess.DEVNULL, universal_newlines=True)
+    except (subprocess.CalledProcessError, OSError):
+        print("Please make sure that the command line tool snapcraft is")
+        print("installed and that you have logged in to an account by running")
+        print("'snapcraft login'.")
+        sys.exit(1)
+
+
+def get_snap_revisions(snap, version):
+    """Finds the revisions for the snap and version in the beta channel.
+
+    If you call this function without being logged in with snapcraft, it
+    will hang with no output.
+
+    :param str snap: the name of the snap on the snap store
+    :param str version: snap version number, e.g. 1.7.0
+
+    :returns: list of revision numbers
+    :rtype: `list` of `str`
+
+    :raises subprocess.CalledProcessError: if the snapcraft command
+        fails
+
+    :raises AssertionError: if the expected snaps are not found
+
+    """
+    cmd = ['snapcraft', 'status', snap]
+    process = subprocess.run(cmd, check=True, stdout=subprocess.PIPE, universal_newlines=True)
```

Comment:
I know it should be covered by running `assert_logged_into_snapcraft` first, but just as a double check, could we print something like "getting snap revisions" here so we know where we've stalled?

Reply:
Good idea! I did this.